src/dapp/components/farm/Gold.tsx

```diff
@@ -69,7 +69,7 @@ export const Gold: React.FC<Props> = ({ inventory }) => {
       load();
       setAmount(0);
     }
-  }, [machineState.value]);
+  }, [machineState, machineState.value]);
```

Comment:
Will the reference of `machineState` change in this case?

Reply:
Oh, local linter must have caught this one. I'll resolve it.

Reply:
@adamhannigan I retained this, as it doesn't appear to be affected erroneously by updates to `machineState`, since it's not a deep check of the parent object. It's proper to include it as, if the reference is no longer valid the `value` child must also have changed.